### ISSUE TITLE
ctest -V fails test 2 - invasive on clean install.

### ISSUE THREAD

**Issue by pireads** [libgit2/libgit2#6263]
### Reproduction steps

1. check out the main branch of libgit2
2. `mkdir build && cd build`
3. `cmake ..`
4. `cmake --build .`
5. `ctest -V`

### Expected behavior

All tests pass.

### Actual behavior

```
test 2
    Start 2: invasive

2: Test command: /Users/avi/asu/cse485/libgit2/build/libgit2_tests "-v" "-score::ftruncate" "-sfilter::stream::bigfile" "-sodb::largefiles" "-siterator::workdir::filesystem_gunk" "-srepo::init" "-srepo::init::at_filesystem_root"
2: Test timeout computed to be: 10000000
2: No suite matching 'core::ftruncate' found.
1/1 Test #2: invasive .........................***Failed    0.00 sec
No suite matching 'core::ftruncate' found.


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.01 sec

The following tests FAILED:
	  2 - invasive (Failed)
Errors while running CTest
```

(after running `ctest -V --rerun-failed --output-on-failure`)

All tests pass with `./libgit2_tests`.

### Version of libgit2 (release number or SHA1)

`2fc0fcb5d6827c4107c571a7a076d029f7d3c442`

### Operating system(s) tested

`Darwin minas-tirith.3521.dhcp.asu.edu 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64` (macos 12.3.1)

`Linux bansheerubber 5.16.15-arch1-1 #1 SMP PREEMPT Thu, 17 Mar 2022 00:30:09 +0000 x86_64 GNU/Linux` (Arch Linux)

**Comment by bansheerubber** [comment]
Running the `core::ftruncate` test seems to imply that there should be a test at `./tests/libgit2/core/ftruncate.c`, but instead that files appears to have been moved as of 06f5d3965c63b83184e4bc40b20bad40fbd55e4d to the `./tests/util/` folder, and now ftruncate seems to be covered during util_tests. When I ran the invasive test with `core::ftruncate` included I also got a failure (I am running the Arch Linux OS in the issue description), but when I run the invasive tests without `core::ftruncate` included all my tests pass. Do any maintainers know if this could be the cause of the problem?

**Comment by ethomson** [comment]
> Running the core::ftruncate test seems to imply that there should be a test at ./tests/libgit2/core/ftruncate.c, but instead that files appears to have been moved as of https://github.com/libgit2/libgit2/commit/0d1745d294177cf66d773bfce8043ae289d86788 to the ./tests/util/ folder, and now ftruncate seems to be covered during util_tests.

Yes, that's correct. I'll fix that. I don't think the maintainers generally use the `ctest` wrapper - at least not to run all the tests. 🤔 

> When I ran the invasive test with core::ftruncate included I also got a failure (I am running the Arch Linux OS in the issue description)

How did they fail?

**Comment by bansheerubber** [comment]
> How did they fail?

In the actual behavior section in the issue OP, the test outputs `No suite matching 'core::ftruncate' found.` and does not complete the invasive tests. `ctest` seems to treat this as a failure, probably because the invasive test command `./libgit2_tests -v -sutil::ftruncate -sfilter::stream::bigfile -sodb::largefiles -siterator::workdir::filesystem_gunk -srepo::init -srepo::init::at_filesystem_root` exits with code 255 instead of 0.
